python/tvm/script/printer/doc.py

```diff
@@ -47,3 +103,171 @@ def __init__(self, value):
             self.__init_handle_by_constructor__(_ffi_api.LiteralDocInt, value)  # type: ignore
         else:
             raise TypeError(f"Unsupported type {type(value)} for LiteralDoc")
+
+
+@tvm._ffi.register_object("script.printer.IdDoc")
+class IdDoc(ExprDoc):
+    """Doc that represents identifier"""
+
+    name: str
+
+    def __init__(self, name: str):
+        self.__init_handle_by_constructor__(_ffi_api.IdDoc, name)  # type: ignore
+
+
+@tvm._ffi.register_object("script.printer.AttrAccessDoc")
+class AttrAccessDoc(ExprDoc):
+    """Doc that represents attribute access on an expression"""
+
+    value: ExprDoc
+    attr: str
+
+    def __init__(self, value: ExprDoc, attr: str):
+        self.__init_handle_by_constructor__(_ffi_api.AttrAccessDoc, value, attr)  # type: ignore
+
+
+@tvm._ffi.register_object("script.printer.IndexDoc")
+class IndexDoc(ExprDoc):
+    """Doc that represents index access on an expression"""
+
+    value: ExprDoc
+    indices: tvm.ir.container.Array  # actual type:  List[Union[ExprDoc, "SliceDoc"]]
```

Comment:
Simply using the actual type

```suggestion
    indices: List[Union[ExprDoc, "SliceDoc"]]
```